[cmd/telemetrygen] expose the generated span duration as a command parameter

jmsnll opened the pull request:
**Description:** 

Exposes the duration of generated spans as a command line parameter.  It uses a `DurationVar` flag so units can be easily provided and are automatically applied.

Example usage:

```bash
telemetrygen traces --traces 100 --otlp-insecure --span-duration 10ns # nanoseconds
telemetrygen traces --traces 100 --otlp-insecure --span-duration 10us # microseconds
telemetrygen traces --traces 100 --otlp-insecure --span-duration 10ms # milliseconds
telemetrygen traces --traces 100 --otlp-insecure --span-duration 10s # seconds
```

**Testing:** 

Ran without the argument provided `telemetrygen traces --traces 1 --otlp-insecure` and seen spans publishing with the default value.

Ran again with the argument provided:  `telemetrygen traces --traces 1 --otlp-insecure --span-duration 1s`

And observed the expected output:

```
Resource SchemaURL: 
Resource attributes:
     -> service.name: Str(telemetrygen)
ScopeSpans #0
ScopeSpans SchemaURL: 
InstrumentationScope telemetrygen 
Span #0
    Trace ID       : 9718172ff37b17389089b99667ae9522
    Parent ID      : 4eb33cd3ed81dc3e
    ID             : 2d768f6dbde63e7c
    Name           : okey-dokey
    Kind           : Server
    Start time     : 2023-09-18 23:33:18.111136 +0000 UTC   <----
    End time       : 2023-09-18 23:33:19.111136 +0000 UTC   <----
    Status code    : Unset
    Status message : 
Attributes:
     -> net.peer.ip: Str(1.2.3.4)
     -> peer.service: Str(telemetrygen-client)
Span #1
    Trace ID       : 9718172ff37b17389089b99667ae9522
    Parent ID      : 
    ID             : 4eb33cd3ed81dc3e
    Name           : lets-go
    Kind           : Client
    Start time     : 2023-09-18 23:33:18.111136 +0000 UTC   <----
    End time       : 2023-09-18 23:33:19.111136 +0000 UTC   <----
    Status code    : Unset
    Status message : 
Attributes:
     -> net.peer.ip: Str(1.2.3.4)
     -> peer.service: Str(telemetrygen-server)
        {"kind": "exporter", "data_type": "traces", "name": "logging"}
```

**Documentation:** No documentation added.

Commit:
[cmd/telemetrygen] expose the generated span duration as a command parameter

Commit:
chore: add chloggen entry

jmsnll committed:
Merge branch 'main' into feat/random-fake-span-durations

Commit:
style: run `gci` tool to fix linting issue

jpkrohling reviewed (commented):
I think we have tests that could be used to exercise this. Can you double-check and add one accordingly?

Commit:
[cmd/telemetrygen] calculate and set span start and end times precisely

- previous approach depended on a second call of `time.Now()` which differed in value from the spans start time and introduced variance to the duration
- both start and end times are now set for both parent and child spans
- added test to ensure start and end times match the duration provided exactly

jmsnll commented:
I will look at this today to get the branch updated
